File: Luxena.Travel/src/Core/Domain/Products/Accommodation.cs
using System;

using Luxena.Domain;


namespace Luxena.Travel.Domain
{

	[RU("Проживание", "Проживания")]
	public partial class Accommodation : Product
	{

		[SemanticSetup]
		public static void AnnotationSetup(SemanticSetup<Accommodation> se)
		{
			se.For(a => a.ReissueFor)
				.Suggest<Accommodation>();

			se.For(a => a.Provider)
				.Suggest<AccommodationProvider>();
		}


		public override ProductType Type => ProductType.Accommodation;

		public override string Name => DomainRes.Accommodation;

		public override string PassengerName => GetPassengerNames();


		[Patterns.StartDate]
		public virtual DateTime StartDate { get; set; }

		[Patterns.FinishDate]
		public virtual DateTime? FinishDate { get; set; }


		[RU("Гостиница")]
		public virtual string HotelName { get; set; }

		[RU("Офис гостиницы", ruShort: "офис")]
		public virtual string HotelOffice { get; set; }

		[RU("Код гостиницы", ruShort: "код")]
		public virtual string HotelCode { get; set; }


		[RU("Расположение")]
		public virtual string PlacementName { get; set; }

		[RU(ruShort: "офис")]
		public virtual string PlacementOffice { get; set; }

		[RU(ruShort: "код")]
		public virtual string PlacementCode { get; set; }


		public virtual AccommodationType AccommodationType { get; set; }

		public virtual CateringType CateringType { get; set; }


		public new partial class Service : Service<Accommodation>
		{

		}

	}

}

File: src.15/Web/app/Products/Accommodation.ts
module Luxena.Views
{
	registerEntityControllers(sd.Accommodation, se => ({
		
		list: [
			se.IssueDate,
			se.Name,
			se.PassengerName,
			se.Provider,
			se.Customer,
			se.Order,
			se.Total,
			se.ServiceFee,
			se.GrandTotal,
		],

		view: {
			"fields1": [
				se.IssueDate,
				se.ReissueFor,

				se.PassengerRow,
				se.CustomerAndOrder,
				se.Intermediary,

				se.Provider,
				se.StartDate,
				se.FinishDate,
				se.Country,
				se.PnrCode,
				se.TourCode,

				se.HotelName, se.HotelOffice, se.HotelCode,
				se.PlacementName, se.PlacementOffice, se.PlacementCode,

				se.AccommodationType,
				se.CateringType,

				se.SellerAndOwner,
			],
			"fields2": se.Finance,
			"fields3": se.Note,
		},

		edit: {
			"fields1": [
				se.IssueDateAndReissueFor,

				se.PassengerRow,
				se.CustomerAndOrder,
				se.Intermediary,

				se.Provider,
				se.StartAndFinishDate,
				se.Country,
				se.PnrAndTourCode,

				Ui.fieldRow(se, se.HotelName, [se.HotelName, se.HotelOffice, se.HotelCode, ]),
				Ui.fieldRow(se, se.PlacementName, [se.PlacementName, se.PlacementOffice, se.PlacementCode, ]),

				Ui.fieldSet(se, null, [se.AccommodationType], [se.CateringType]),

				se.SellerAndOwner,
			],
			"fields2": se.Finance,
			"fields3": se.Note,
		}
	}));

}